Flaking unit test: User Defined Network Controller manager reconcile UDN CR [It] when UDN is being deleted, NAD exist, 2 pods using UDN, should delete NAD once no pod uses the network

### Which jobs are flaking?

Unit test
 https://github.com/ovn-kubernetes/ovn-kubernetes/actions/runs/11842209785/job/33000292040?pr=4837

### Which tests are flaking?

 `User Defined Network Controller manager reconcile UDN CR [It] when UDN is being deleted, NAD exist, 2 pods using UDN, should delete NAD once no pod uses the network`

### Since when has it been flaking?

I first saw it yesterday (November 14th 2024)

### Reason for failure (if possible)

I've seen this unit test flake a couple of times already:

```
2024-11-14T17:23:54.2550079Z Summarizing 1 Failure:
2024-11-14T17:23:54.2552251Z   [FAIL] User Defined Network Controller manager reconcile UDN CR [It] when UDN is being deleted, NAD exist, 2 pods using UDN, should delete NAD once no pod uses the network
2024-11-14T17:23:54.2555077Z   /home/runner/work/ovn-kubernetes/ovn-kubernetes/go-controller/pkg/clustermanager/userdefinednetwork/controller_test.go:1278
2024-11-14T17:23:54.2556103Z 
2024-11-14T17:23:54.2556529Z Ran 56 of 56 Specs in 16.599 seconds
2024-11-14T17:23:54.2557879Z FAIL! -- 55 Passed | 1 Failed | 0 Pending | 0 Skipped
2024-11-14T17:23:54.2559128Z --- FAIL: TestUserDefinedNetworkController (16.65s)
```
For example: https://github.com/ovn-kubernetes/ovn-kubernetes/actions/runs/11842209785/job/33000292040?pr=4837

In particular:
```
 ------------------------------
2024-11-14T17:23:40.7725384Z User Defined Network Controller manager reconcile UDN CR when UDN is being deleted, NAD exist, 2 pods using UDN, should delete NAD once no pod uses the network
2024-11-14T17:23:40.7727202Z /home/runner/work/ovn-kubernetes/ovn-kubernetes/go-controller/pkg/clustermanager/userdefinednetwork/controller_test.go:324
2024-11-14T17:23:40.7737472Z I1114 17:23:40.773543   31971 factory.go:432] Starting watch factory
2024-11-14T17:23:41.0772620Z I1114 17:23:41.076350   31971 controller.go:140] Starting user-defined network controllers
2024-11-14T17:23:41.0774302Z I1114 17:23:41.076422   31971 controller.go:103] Adding controller udn-nad-controller event handlers
2024-11-14T17:23:41.0776244Z I1114 17:23:41.076458   31971 controller.go:103] Adding controller cluster-user-defined-network-controller event handlers
2024-11-14T17:23:41.0778188Z I1114 17:23:41.076482   31971 controller.go:103] Adding controller user-defined-network-controller event handlers
2024-11-14T17:23:41.0779969Z I1114 17:23:41.076517   31971 controller.go:103] Adding controller udn-namespace-controller event handlers
2024-11-14T17:23:41.0781642Z I1114 17:23:41.076493   31971 shared_informer.go:313] Waiting for caches to sync for udn-nad-controller
2024-11-14T17:23:41.0783349Z I1114 17:23:41.076567   31971 shared_informer.go:313] Waiting for caches to sync for udn-namespace-controller
2024-11-14T17:23:41.0784957Z I1114 17:23:41.076572   31971 shared_informer.go:320] Caches are synced for udn-nad-controller
2024-11-14T17:23:41.0786492Z I1114 17:23:41.076587   31971 shared_informer.go:320] Caches are synced for udn-namespace-controller
2024-11-14T17:23:41.0788224Z I1114 17:23:41.076628   31971 shared_informer.go:313] Waiting for caches to sync for user-defined-network-controller
2024-11-14T17:23:41.0790092Z I1114 17:23:41.076647   31971 shared_informer.go:320] Caches are synced for user-defined-network-controller
2024-11-14T17:23:41.0791887Z I1114 17:23:41.076704   31971 shared_informer.go:313] Waiting for caches to sync for cluster-user-defined-network-controller
2024-11-14T17:23:41.0792999Z I1114 17:23:41.076725   31971 shared_informer.go:320] Caches are synced for cluster-user-defined-network-controller
2024-11-14T17:23:41.0794353Z I1114 17:23:41.076755   31971 controller.go:127] Starting controller cluster-user-defined-network-controller with 1 workers
2024-11-14T17:23:41.0795453Z I1114 17:23:41.076789   31971 controller.go:127] Starting controller user-defined-network-controller with 1 workers
2024-11-14T17:23:41.0796427Z I1114 17:23:41.076832   31971 controller.go:127] Starting controller udn-nad-controller with 1 workers
2024-11-14T17:23:41.0797366Z I1114 17:23:41.076865   31971 controller.go:127] Starting controller udn-namespace-controller with 1 workers
2024-11-14T17:23:41.0798203Z   STEP: mark UDN for deletion @ 11/14/24 17:23:41.077
2024-11-14T17:23:41.0798952Z I1114 17:23:41.077543   31971 controller.go:398] Finalizer removed from UserDefinedNetworks [test/test]
2024-11-14T17:23:41.0799756Z I1114 17:23:41.078912   31971 controller.go:447] Updated status UserDefinedNetwork [test/test]
2024-11-14T17:23:42.0795408Z   [FAILED] in [It] - /home/runner/work/ovn-kubernetes/ovn-kubernetes/go-controller/pkg/clustermanager/userdefinednetwork/controller_test.go:1278 @ 11/14/24 17:23:42.078
2024-11-14T17:23:42.0798231Z I1114 17:23:42.079474   31971 factory.go:548] Stopping watch factory
2024-11-14T17:23:42.0804294Z • [FAILED] [1.307 seconds]
2024-11-14T17:23:42.0806416Z User Defined Network Controller manager reconcile UDN CR [It] when UDN is being deleted, NAD exist, 2 pods using UDN, should delete NAD once no pod uses the network
2024-11-14T17:23:42.0809259Z /home/runner/work/ovn-kubernetes/ovn-kubernetes/go-controller/pkg/clustermanager/userdefinednetwork/controller_test.go:324
2024-11-14T17:23:42.0810533Z 
2024-11-14T17:23:42.0810916Z   [FAILED] Timed out after 1.001s.
2024-11-14T17:23:42.0811491Z   Expected
2024-11-14T17:23:42.0811980Z       <[]v1.Condition | len:1, cap:1>: [
2024-11-14T17:23:42.0812607Z           {
2024-11-14T17:23:42.0813129Z               Type: "NetworkReady",
2024-11-14T17:23:42.0813778Z               Status: "True",
2024-11-14T17:23:42.0814412Z               ObservedGeneration: 0,
2024-11-14T17:23:42.0815095Z               LastTransitionTime: {
2024-11-14T17:23:42.0815973Z                   Time: 0001-01-01T00:00:00Z,
2024-11-14T17:23:42.0816544Z               },
2024-11-14T17:23:42.0817288Z               Reason: "NetworkAttachmentDefinitionReady",
2024-11-14T17:23:42.0818389Z               Message: "NetworkAttachmentDefinition has been created",
2024-11-14T17:23:42.0819108Z           },
2024-11-14T17:23:42.0819467Z       ]
2024-11-14T17:23:42.0819804Z   to equal
2024-11-14T17:23:42.0820302Z       <[]v1.Condition | len:1, cap:1>: [
2024-11-14T17:23:42.0820850Z           {
2024-11-14T17:23:42.0821379Z               Type: "NetworkReady",
2024-11-14T17:23:42.0822023Z               Status: "False",
2024-11-14T17:23:42.0822696Z               ObservedGeneration: 0,
2024-11-14T17:23:42.0823415Z               LastTransitionTime: {
2024-11-14T17:23:42.0824321Z                   Time: 0001-01-01T00:00:00Z,
2024-11-14T17:23:42.0824951Z               },
2024-11-14T17:23:42.0825517Z               Reason: "SyncError",
2024-11-14T17:23:42.0827709Z               Message: "failed to delete NetworkAttachmentDefinition [test/test]: network in use by the following pods: [test/pod-1 test/pod-2]",
2024-11-14T17:23:42.0828970Z           },
2024-11-14T17:23:42.0829420Z       ]
2024-11-14T17:23:42.0831615Z   In [It] at: /home/runner/work/ovn-kubernetes/ovn-kubernetes/go-controller/pkg/clustermanager/userdefinednetwork/controller_test.go:1278 @ 11/14/24 17:23:42.078
2024-11-14T17:23:42.0833666Z ------------------------------

```


### Anything else we need to know?

_No response_

--- a/go-controller/pkg/clustermanager/userdefinednetwork/controller_test.go
+++ b/go-controller/pkg/clustermanager/userdefinednetwork/controller_test.go
@@ -325,25 +325,16 @@ var _ = Describe("User Defined Network Controller", func() {
 				var err error
 				nad := testNAD()
 				udn := testUDN()
-				c = newTestController(renderNadStub(nad), udn)
-				Expect(c.Run()).To(Succeed())
+				udn.SetDeletionTimestamp(&metav1.Time{Time: time.Now()})
 
 				testOVNPodAnnot := map[string]string{util.OvnPodAnnotationName: `{"default": {"role":"primary"}, "test/test": {"role": "secondary"}}`}
 				pod1 := &corev1.Pod{ObjectMeta: metav1.ObjectMeta{Name: "pod-1", Namespace: udn.Namespace, Annotations: testOVNPodAnnot}}
-				pod1, err = cs.KubeClient.CoreV1().Pods(udn.Namespace).Create(context.Background(), pod1, metav1.CreateOptions{})
-				Expect(err).NotTo(HaveOccurred())
 				pod2 := &corev1.Pod{ObjectMeta: metav1.ObjectMeta{Name: "pod-2", Namespace: udn.Namespace, Annotations: testOVNPodAnnot}}
-				pod2, err = cs.KubeClient.CoreV1().Pods(udn.Namespace).Create(context.Background(), pod2, metav1.CreateOptions{})
-				Expect(err).NotTo(HaveOccurred())
 
-				By("mark UDN for deletion")
-				udn.SetDeletionTimestamp(&metav1.Time{Time: time.Now()})
-				udn, err = cs.UserDefinedNetworkClient.K8sV1().UserDefinedNetworks(udn.Namespace).Update(context.Background(), udn, metav1.UpdateOptions{})
-				Expect(err).NotTo(HaveOccurred())
-				Expect(udn.DeletionTimestamp.IsZero()).To(BeFalse())
-
+				c = newTestController(renderNadStub(nad), udn, nad, pod1, pod2)
 				// user short interval to make the controller re-enqueue requests
 				c.networkInUseRequeueInterval = 50 * time.Millisecond
+				Expect(c.Run()).To(Succeed())
 
 				assertFinalizersPresent(cs.UserDefinedNetworkClient, cs.NetworkAttchDefClient, udn, pod1, pod2)
 

